### PR TITLE
input_chunk: handle filter_do edge case

### PR DESCRIPTION
**Pull request by braydonk**
`flb_filter_do` may modify the input chunk's total records, meaning that if there is a filter in the pipeline the total records could be double-counted which breaks anything that relies on an event chunk's `total_records`.

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:

- [ ] Example configuration file for the change
- [ ] Debug log output from testing the change

- [ ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.

- [ ] Run [local packaging test](./packaging/local-build-all.sh) showing all targets (including any new ones) build.
- [ ] Set `ok-package-test` label to test for all targets (requires maintainer to do).

**Documentation**

- [ ] Documentation required for this feature

**Backporting**

- [ ] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

**Comment by braydonk** [comment]
I tried to add a test for this similar to my last PR, but unfortunately the test's mechanism for reading total records doesn't have enough granularity to get exact record counts and make sure that it's a precise expected number.

File: src/flb_input_chunk.c [hunk 1]
```diff
@@ -1495,6 +1495,15 @@ static int input_chunk_append_raw(struct flb_input_instance *in,
         pre_real_size = flb_input_chunk_get_real_size(ic);
     }
 
+    /*
+     * Set the total_records based on the records that n_records
+     * says we should be writing. This may be modified by flb_filter_do,
+     * where a filter may add/remove records.
```
Comment (ridwanmsharif) [inline src/flb_input_chunk.c:1501]: This only works because `flb_filter_do` **overwrites** the total records and not adds on it right? Maybe we drop a comment about that?

Saying that cause I had to stare at this change for too long to uderstand how this move avoids the double write

Comment (braydonk) [reply src/flb_input_chunk.c]: Done in [8543d12](https://github.com/fluent/fluent-bit/pull/8229/commits/8543d126d287217567af15d16fdd0e42a56c71a3)

**Comment by braydonk** [comment]
macos unit test failures are flakes.

**Comment by braydonk** [comment]
@leonardo-albertovich bumping this, fixes an edge case I missed in the last PR.
